src/pages/transfer/transfer.ts

```diff
@@ -14,7 +22,137 @@ import { IonicPage, NavController, NavParams } from "ionic-angular";
   templateUrl: "transfer.html",
 })
 export class TransferPage {
+  fromWallet: IWallet;
+  toPublicKey: string;
+  balances: IBalance[];
+  chosenCurrency: string;
+  chosenCurrencyAmount: number;
+  amount: number;
+  errorMessage: string;
+  successMessage: string;
+  enoughFunds: boolean;
+  password: string;
+  transferButtonEnabled: boolean;
 
-  constructor(public navCtrl: NavController, public navParams: NavParams) {}
+  constructor(private navParams: NavParams,
+              private transactionSignService: TransactionSignService,
+              private transferTransactionService: TransferTransactionService) {}
 
+  ionViewDidLoad(): void {
+    this.transferButtonEnabled = true;
+    this.fromWallet = this.navParams.get("currentWallet");
+    this.balances = this.navParams.get("currentWalletBalance");
+    for (let currency of this.balances) { // DELETE later just for testing
+      let random = Math.floor(Math.random() * 9999); 
+      currency.amount = random;
+    }
+    this.chosenCurrency = this.balances[0].currency;
+    this.chosenCurrencyAmount = this.balances[0].amount;
+  }
+
+  setChosenCurrencyAmount(): boolean {
+    for (let currency of this.balances) {
+      if (currency.currency === this.chosenCurrency) {
+        this.chosenCurrencyAmount = currency.amount;
+        this.onAmountChanged();
+        return true;
+      }
+    }
+    return false;
+  }
+
+  canTransfer(): boolean {
+    if (this.toPublicKey === undefined) {
+      this.errorMessage = "The publickey to send is not defined";
+      return false;
+    } else if (this.amount === undefined) {
+      this.errorMessage = "The amount is not defined";
+      return false;
+    } else if (!this.enoughFunds) {
+      this.errorMessage = "Not enough funds";
+      return false;
+    } else if (this.toPublicKey === this.fromWallet.publicKey) {
+      this.errorMessage = "Can't send to own wallet";
+      return false;
+    } else if (this.password === "") {
+      this.errorMessage = "Password field is empty";
+      return false;
+    } else {
+      this.errorMessage = "";
+      return true;
+    }
+  }
+
+  resetTransferState() {
+    this.errorMessage = "";
+    this.successMessage = "";
+  }
+
+  onAmountChanged(): void {
+    this.resetTransferState();
+    if (this.amount === undefined || this.amount.toString() === "") {
+      this.enoughFunds = undefined;
+    } else if (this.amount <= this.chosenCurrencyAmount) {
+      this.enoughFunds = true;
+    } else {
+      this.enoughFunds = false;
+    }
+  }
+
+  transfer(): void {
+    this.resetTransferState();
+    this.transferButtonEnabled = false;
+    if(this.canTransfer()) {
+      this.signTransaction();
+    } else {
+      this.transferButtonEnabled = true;
+    }
+  }
+
+  signTransaction(): void {
```

Comment:
The name of this method does not full cover what is actually done. As far as I can see this function is responsible for:
- Building the transaction object
- Signing the transaction
- Sending the transaction to the API (by calling the `sendTransactions` method).

Maybe it would be nicer if this function was solely responsible for signing the transactions. Then there could be another function which would create the Transaction object.

The `transfer` method would then call the `createTransaction`, `signTransaction` and `sendTransaction` methods.

By trying to make methods do one thing you also make it easier to unit test them.